### ISSUE TITLE
Derived symbol value is empty when it is in Symbol geneator

### ISSUE THREAD

**Issue by chunyu3** [dotnet/templating#6291]
in dotnet 7.0.2, When we reference a derived symbol in a symbol generator, the generated symbol value does not contain the derived symbol.
run 
`dotnet new *** --ClientName test`

The value of  `PackageName` : 
Actual: `Package.`
expected:  `Package.Test`

It works fine in dotnet 3.1.

```
"symbols": {
"clientName": {
      "type": "parameter",
      "datatype": "text",
      "isRequired": true,
      "description": "client name"
    },
"ClientNameCapitalCase": {
      "type": "derived",
      "datatype": "text",
      "valueSource": "clientName",
      "valueTransform": "firstUpperCase",
      "replaces": "Template"
    },
"PackageName": {
      "type": "generated",
      "generator": "join",
      "parameters": {
        "symbols": [
          {
            "type": "const",
            "value": "Package"
          },
          {
            "type": "ref",
            "value": "ClientNameCapitalCase"
          }
        ],
        "separator": "."
      },
    },
}
```

**Comment by YuliiaKovalova** [comment]
Hi @chunyu3,

Thank you for reporting the bug.
We are investigating the issue.

**Comment by GangWang01** [comment]
It was reproduced with dotnet 7.0.2. Root cause is the wrong order processing generated symbol and computed symbol at https://github.com/dotnet/templating/blob/a51479c5329355d4993486750d5c56c6b32bfdaf/src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs#L376-L387. Release/7.0.3xx should also have this problem.

This issue was fixed in PR https://github.com/dotnet/templating/pull/5223 that refactored at https://github.com/dotnet/templating/blob/3ea9033de81a08f91ae513c649af1f3b3c8bbcd7/src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/MacroProcessor.cs#L27-L79. dotnet 8.0.100-preview.2 has the fix.

BTW, .NET 6.0 doesn't have this problem.

**Comment by YuliiaKovalova** [comment]
Hi @chunyu3,

Could you specify the urgency of this fix for your tasks? Do you have a workaround?

**Comment by chunyu3** [comment]
Thanks @YuliiaKovalova  and @GangWang01 for the quick resolve. Yes, it will block our project to generate the correct project via template and it has current no workaround now. 
@YuliiaKovalova , when can I get a release version contains this resolution? Thanks

**Comment by YuliiaKovalova** [comment]
Hi @chunyu3,

We are discussing possible solutions behind the scene. I will give you more information once know more.

**Comment by YuliiaKovalova** [comment]
Hi @chunyu3 ,

The issue is fixed for release/7.0.2xx. 
Please expect to receive these changes with the next patch  - preliminary in the middle of June.
Thank you for your patience!

Best Regards,
Yuliia Kovalova

**Comment by chunyu3** [comment]
> 7.0.2

Thanks for your help!
